packages/@aws-cdk/core/lib/assets.ts

```diff
@@ -174,6 +174,19 @@ export interface DockerImageAssetSource {
    */
   readonly dockerBuildArgs?: { [key: string]: string };
 
+  /**
+   * Build secrets to pass to the `docker build` command.
+   *
+   * Since Docker build secrets are resolved before deployment, keys and
+   * values cannot refer to unresolved tokens (such as `lambda.functionArn` or
+   * `queue.queueUrl`).
+   *
+   * Only allowed when `directoryName` is specified.
```

Comment:
Could get a unit test that verifies that an error is thrown when `directoryName` is unspecified?

Reply:
Honestly, I copied from the existing `dockerBuildArgs` and `dockerBuildTarget` which don't appear to have unit tests.

Furthermore, I don't believe this functions are used by end users, rather internally within the CDK.

May I suggest that logic and supporting tests could be implemented as part of a separate issue?

Reply:
They are tested but only in a deprecated unit test. Could you add both `dockerBuildArgs` and `dockerBuildSecrets` to a new unit test below this one? https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/core/test/stack-synthesis/new-style-synthesis.test.ts#L212

Reply:
Great, I missed that.  Done.